src/sentry/identity/oauth2.py

```diff
@@ -30,24 +31,92 @@ class OAuth2Provider(Provider):
 
     oauth_scopes = ()
 
+    def get_oauth_access_token_url(self):
```

Comment:
Let's add a helper method to do something like this.

Also see my comment below about reading the values from the config.

```py
def _get_oauth_parameter(self, parameter_name):
    """
    Lookup an OAuth parameter for the provider. Depending on the context of the
    pipeline using the provider, the parameter may come from 1 of 3 places:

    1. Check the class property of the provider for the parameter.

    2. If the provider has the parameters made available within the ``config``.

    3. If provided, check the pipeline's ``provider_model`` for the oauth parameter
       in the config field.

    If the parameter cannot be found a KeyError will be raised.
    """
    prop = getattr(self, u'oauth_{}'.format(parameter_name))
    if prop is not '':
        return prop

    if self.config.get(parameter_name):
        return self.config.get(parameter_name)

    model = self.pipeline.provider_model
    if model and model.config.get(parameter_name) is not None:
        return model.config.get(parameter_name)

    raise KeyError
```

then all of these methods just become like

```py
def get_oauth_access_token_url(self):
    return self._get_oauth_parameter('access_token_url')
```